lib/roll.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
```

Comment:
Again, very impressed with the attention to detail in dealing with these edge cases